### PR TITLE
added new linter gomodguard

### PR DESCRIPTION
**Pull request by ryancurrah**
added a new linter gomodguard, disabled by default

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -488,6 +489,7 @@ golangci-lint help linters
 - [wsl](https://github.com/bombsimon/wsl) - Whitespace Linter - Forces you to use empty lines!
 - [goprintffuncname](https://github.com/jirfag/go-printf-func-name) - Checks that printf-like functions are named with `f` at the end
 - [gomnd](https://github.com/tommy-muehle/go-mnd) - An analyzer to detect magic numbers.
+- [gomodguard](https://github.com/rcurrah/gomodguard) - Allow and block list linter for direct Go module dependencies.
```
Comment (hairyhenderson) [inline README.md:492]:
```suggestion
- [gomodguard](https://github.com/ryancurrah/gomodguard) - Allow and block list linter for direct Go module dependencies.
```

File: README.md [hunk 2]
```diff
@@ -1261,6 +1275,7 @@ Thanks to developers and authors of used linters:
 - [bombsimon](https://github.com/bombsimon)
 - [jirfag](https://github.com/jirfag)
 - [tommy-muehle](https://github.com/tommy-muehle)
+- [rcurrah](https://github.com/rcurrah)
```
Comment (hairyhenderson) [inline README.md:1278]:
```suggestion
- [ryancurrah](https://github.com/ryancurrah)
```

Comment (ryancurrah) [reply README.md]: Fixed

File: README.md (hunk 1 above)
Comment (ryancurrah) [reply README.md]: Fixed

**Comment by ryancurrah** [comment]
Looks like the build failed for no reason. If someone could re-launch it that would be great. Otherwise I'll just push a superfluous commit.

**Comment by ryancurrah** [comment]
Done

**Pull request merged**